File: app/src/main/java/com/onix/streamer/demo/actioncamera/cam/GoPro3Hero.java
package com.onix.streamer.demo.actioncamera.cam;

import android.text.TextUtils;

import com.onix.streamer.demo.actioncamera.cam.base.GoProBase;
import com.onix.streamer.demo.util.IoUtil;

public class GoPro3Hero extends GoProBase {

    private static String CMD_START_PREVIEW = "http://10.5.5.9/camera/PV?t=onixgopro&p=%02";
    private static String PREVIEW_URL = "http://10.5.5.9:8080/live/amba.m3u8";

    @Override
    public String getPreviewUrl() {
        if (!TextUtils.isEmpty(IoUtil.getSyncHttp(CMD_START_PREVIEW))) {
            return PREVIEW_URL;
        } else {
            return null;
        }
    }
}

File: app/src/main/java/com/onix/avlib/demo/action_camera/cam/GoPro3Hero.kt
package com.onix.avlib.demo.action_camera.cam

import android.text.TextUtils

import com.onix.avlib.demo.action_camera.cam.base.GoProBase
import com.onix.avlib.demo.util.IoUtil


class GoPro3Hero : GoProBase() {

    override fun getPreviewUrl(): String? {
        return if (!TextUtils.isEmpty(IoUtil.getSyncHttp(CMD_START_PREVIEW))) {
            PREVIEW_URL
        } else {
            null
        }
    }

    companion object {

        private val CMD_START_PREVIEW = "http://10.5.5.9/camera/PV?t=onixgopro&p=%02"
        private val PREVIEW_URL = "http://10.5.5.9:8080/live/amba.m3u8"
    }
}
